src/BenchmarksServer/Startup.cs

```diff
@@ -211,18 +210,126 @@ private static string CloneAndRestore(string path, ServerJob job)
 
             Debug.Assert(benchmarksDir != null);
 
-            // TODO: add project references to checked out repos
-            // https://github.com/aspnet/benchmarks/issues/185
+            var builtDirs = new List<string>();
+            var builtPackages = new List<BuiltPackage>();
 
-            // Restore in each dir
-            foreach (var dir in dirs)
+            foreach (var dir in dirs.Except(new[] { benchmarksDir }))
             {
+                UpdateNuGetConfig(path, dir, builtDirs);
+                UpdateDependencies(path, dir, builtPackages);
+
                 ProcessUtil.Run("dotnet", "restore", workingDirectory: Path.Combine(path, dir));
+                ProcessUtil.Run("cmd", "/c build.cmd build-pack", workingDirectory: Path.Combine(path, dir),
```

Comment:
This is building the package in debug mode which can hurt perf greatly. 3x in the case of Kestrel. @pranavkm says you can change this to "build.cmd --for-ci build-pack?